src/components/TriggerEditor/index.tsx

```diff
@@ -200,22 +202,25 @@ export default ({
 
   const handleFetchInterfacesForTrigger = useCallback(
     async (trigger: AstarteTrigger) => {
-      await handleFetchInterfacesName();
+      const names = await handleFetchInterfacesName();
       const interfaceName = _.get(trigger, 'simpleTriggers[0].interfaceName') as string | undefined;
       if (!interfaceName || interfaceName === '*') {
         return trigger;
       }
-      const ifaceMajors = await handleFetchInterfaceMajors(interfaceName);
-      let ifaceMajor: number | undefined = _.get(trigger, 'simpleTriggers[0].interfaceMajor');
-      if (ifaceMajor == null) {
-        if (ifaceMajors.length === 0) {
-          return trigger;
+      if (interfaceName in names) {
+        const ifaceMajors = await handleFetchInterfaceMajors(interfaceName);
+        let ifaceMajor: number | undefined = _.get(trigger, 'simpleTriggers[0].interfaceMajor');
+        if (ifaceMajor == null) {
+          if (ifaceMajors.length === 0) {
+            return trigger;
+          }
+          ifaceMajor = Math.max(...ifaceMajors);
+          _.set(trigger as AstarteTrigger, 'simpleTriggers[0].interfaceMajor', ifaceMajor);
         }
-        ifaceMajor = Math.max(...ifaceMajors);
-        _.set(trigger as AstarteTrigger, 'simpleTriggers[0].interfaceMajor', ifaceMajor);
+
+        const interfaceMajor = ifaceMajor;
+        await handleFetchInterface({ interfaceName, interfaceMajor });
```

Comment:
At first glance, this part does not affect the json editor. Is there a reason why we're changing it?

Reply:
The handleFetchInterfacesName function triggers on every change when manually filling out the JSON, then it tries to find the interfaceName. Of course, it won't find it until we manually enter it completely and correctly. With this, I have avoided displaying those errors. However, I did this temporarily, and I wanted to check with you if we could come up with a better solution. The image shows the error during the process of finding the matching interfaceName in this case when i dont use  `const names = await handleFetchInterfacesName();

![Screenshot from 2024-07-12 09-57-18](https://github.com/user-attachments/assets/ab814e9e-85f9-4a71-a215-70bda424b192)